### PR TITLE
Ds/python advec1d

### PR DESCRIPTION
**Pull request by dsteinmo**
First serious try at a solver using the python3 API.

### CODE REVIEW

File: advec1d.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    mapI = nodes1d.mapI
+    mapO = nodes1d.mapO
+    nx = nodes1d.nx
+    rx = nodes1d.rx
+    Dr = nodes1d.Dr
+    Lift = nodes1d.Lift
+    Fscale = nodes1d.Fscale
+
+    uC = np.squeeze(np.reshape(u, (u.shape[0]*u.shape[1],1), order='F'), axis=1)
+    nxC = np.squeeze(np.reshape(nx, (nx.shape[0]*nx.shape[1],1), order='F'), axis=1)
```
Comment (mdunphy) [inline advec1d.py]: This seems complicated (error prone?) - if we can find a better way to do this then we'll scare off fewer users

File: advec1d.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    RHS = 0*vmapM
```
Comment (mdunphy) [inline advec1d.py:24]: RHS = np.zeros(vmapM.shape) ?

File: advec1d.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+K = 60
+CFL = 0.25
+
+nodes1d = pyblitzdg.Nodes1DProvisioner(N, K, xmin, xmax)
+nodes1d.buildNodes()
+nodes1d.computeJacobian()
+x = nodes1d.xGrid
+
+u = np.exp(-10*(x*x))
+xplot = np.squeeze(np.reshape(x, ((N+1)*K, 1)))
```
Comment (mdunphy) [inline advec1d.py:61]: Same as comment as at line 22 - seems more complex then it could/should be

**Review by mdunphy — COMMENTED** [review]
General comment: I didn't get through getting it to compile yet so I may be a bit lost on the big picture. Are all of the arrays in the python solver numpy arrays? Do we use any blitz functionality in the Python side? Is this a second project ("PyDG") that uses blitzdg as a dependency?

File: advec1d.py (hunk 1 above)
Comment (dsteinmo) [reply advec1d.py]: Yeah, we could add a helper that does this. Basically all it's doing is reshaping from a 2D tensor to a 1D array and traversing the tensor in column-major (Fortran-style) order.

File: advec1d.py (hunk 2 above)
Comment (dsteinmo) [reply advec1d.py:24]: Sure, I'll change. Wasn't sure what was the preferred way to write it when doing numpy things.

**Comment by dsteinmo** [comment]
@mdunphy My thinking was that any math/flops that happens on the Python side would be done with numpy, since it is more 'python-native' than writing our own wrappers. blitz++ is still being used under the hood to construct the geometry, mapping, differentiation matrices, etc.  I've opted not to expose an API to blitz++ since it seems like a monumental task that would be not unlike rewriting NumPy arrays and arithmetic.

Secondly, I've opted for a 'mono-repo' style-approach for the blitzdg sub-projects for now. Arguably, you can say the 'core' library lives in libblitzdg.(so|a) (one repo), the C++ solvers link to it (maybe the same repo), and so does the pyblitzdg.so wrapper/bindings library (a different repo).

If you think it'll be easier for newcomers to understand, I can split the pyblitzdg bindings code and python script examples to another repo within the WQCG org and just treat the blitzdg 'core' repo as a git submodule. I'm happy to hear your thoughts on this!

File: advec1d.py (hunk 3 above)
Comment (dsteinmo) [reply advec1d.py:61]: Will look into getting a vector from a 2D array.

Comment (dsteinmo) [reply advec1d.py:61]: looks like we probably want this: https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.flatten.html

File: advec1d.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+plt.show()
+while t < finalTime:
+
+    # Calculate Right-hand side.
+    RHS = advec1dComputeRHS(u,c, nodes1d)
+
+    # Update solution.
+    u += dt*RHS
+
+    u_max = np.max(np.max(np.abs(u)))
```
Comment (mdunphy) [inline advec1d.py]: Don't need np.max twice here

File: advec1d.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+x = nodes1d.xGrid
+
+u = np.exp(-10*(x*x))
+xplot = np.squeeze(np.reshape(x, ((N+1)*K, 1)))
+uplot = np.squeeze(np.reshape(u, ((N+1)*K, 1)))
+
+xplot = np.squeeze(np.reshape(x, ((N+1)*K, 1)))
+uplot = np.squeeze(np.reshape(u, ((N+1)*K, 1)))
+
+RHS = 0*u
```
Comment (mdunphy) [inline advec1d.py]: Don't need this; RHS is replaced each trip through your while loop

File: advec1d.py (hunk 2 above)
Comment (mdunphy) [reply advec1d.py:24]: Actually this line isn't even needed, RHS is replaced at line 35

File: advec1d.py (hunk 3 above)
Comment (mdunphy) [reply advec1d.py:61]: Yep that's a big improvement

**Pull request merged**